xapian-letor/feature/tfdoclenfeature.cc

```diff
@@ -41,9 +40,9 @@ TfDoclenFeature::get_values() const
 {
     LOGCALL(API, std::vector<double>, "TfDoclenFeature::get_values", NO_ARGS);
 
-    Query query = Feature::internal->feature_query;
-    map<string, long int> tf = Feature::internal->termfreq();
-    map<string, long int> doc_len = Feature::internal->doc_length();
+    Query query = Feature::feature_query;
+    map<string, long int> tf = Feature::termfreq;
```

Comment:
1. Do you need to scope this to `Feature::` to get it to compile?
2. You've copied this map from `FeatureList::Internal` object to `Feature` object, and then you're copying it again within this method. We can at least eliminate one of these copies, if not both by judicious use of references.